File: src/GeldApp2.Database/Expense.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace GeldApp2.Database
{
    public class Expense
    {
        public Expense()
        {
        }

        public Expense(decimal amount, string category, string subcategory)
            : this(amount, category, subcategory, string.Empty)
        {
        }

        public Expense(decimal amount, string category, string subcategory, string details)
        {
            this.Amount = amount;
            this.Category = category;
            this.Subcategory = subcategory;
            this.Details = details;
            this.Type = amount < 0 ? ExpenseType.Expense : ExpenseType.Revenue;
            this.Created = DateTimeOffset.Now;
            this.Date = this.Created.Date;
            this.LastModified = this.Created;
        }

        public long Id { get; set; }

        public long AccountId { get; set; }

        public string Category { get; set; }

        public string Subcategory { get; set; }

        public string Details { get; set; }

        public decimal Amount { get; set; }

        public DateTime Date { get; set; }

        public ExpenseType Type { get; set; }

        public DateTimeOffset Created { get; set; }

        public string CreatedBy { get; set; }

        public DateTimeOffset LastModified { get; set; }

        public string LastModifiedBy { get; set; }

        /* Navigation properties */

        public Account Account { get; set; }
    }
}

File: src/GeldApp2/ClientApp/src/app/api/model/expense.ts
import { ExpenseType } from './expense-type';

export class Expense {

  id = 0;

  accountName = '';

  amount = 0.0;

  categoryName = '';

  subcategoryName = '';

  details = '';

  // Date as ISO string.
  date = '';

  type: ExpenseType = ExpenseType.expense;

  // Date as ISO string.
  created = '';

  createdBy = '';

  // Date as ISO string.
  lastModified = '';

  lastModifiedBy = '';

  // Defined in the CreateExpenseCommand.
  handlesImportedExpenseId: number | undefined;
}
